### ISSUE TITLE
DOC: `DataFrame.select_dtypes` for non-object string types

### ISSUE THREAD

**Issue by s-banach** [pandas-dev/pandas#54898]
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.select_dtypes.html

### Documentation problem

I want to select pyarrow string columns.
The documentation says to ask for object dtypes, which will not capture pyarrow strings.

Actually, it may be a bug rather than a documentation problem.
If I try something like this:
```python
import pyarrow as pa
df.select_dtypes(include=[pd.ArrowDtype(pa.string()), pd.ArrowDtype(pa.large_string())])
```
Then I get the following error:
`TypeError("string dtypes are not allowed, use 'object' instead")`

### Suggested fix for documentation

Just put this on the list of things that have to be done before pandas 3.0.

**Comment by phofl** [comment]
Yeah select dtypes doesn't work great for EA dtypes, thanks for reporting
